packages/synapse-react-client/src/lib/containers/auth/TwoFactorAuthSettingsPanel.tsx

```diff
@@ -0,0 +1,100 @@
+import ConditionalWrapper from '../utils/ConditionalWrapper'
+import { displayToast } from '../ToastMessage'
+
+export type TwoFactorAuthSettingsPanelProps = {
+  onRegenerateBackupCodes: () => void
+  onBeginTwoFactorEnrollment: () => void
+}
+
+export default function TwoFactorAuthSettingsPanel(
+  props: TwoFactorAuthSettingsPanelProps,
+) {
+  const { onRegenerateBackupCodes, onBeginTwoFactorEnrollment } = props
+  const { data: status, isLoading: isLoadingStatus } =
+    useGetTwoFactorEnrollmentStatus()
+  const isActivated = status?.status === 'ENABLED'
+
+  const { mutate: disable2FA, isLoading: isLoadingMutation } =
+    useDisableTwoFactorAuth({
+      onSuccess: () => {
+        displayToast('2FA removed from this account', 'info')
+      },
+    })
+
+  const isLoading = isLoadingStatus || isLoadingMutation
+
+  return (
+    <Box>
+      <Typography variant={'headline2'}>
+        Two-factor Authentication (2FA)
+      </Typography>
+      <ConditionalWrapper condition={isLoading} wrapper={Skeleton}>
+        <Typography
+          variant={'body1'}
+          color={isActivated ? 'success.main' : 'error.main'}
+          sx={{ my: 2, display: 'flex', alignItems: 'center' }}
+        >
+          <IconSvg
+            icon={isActivated ? 'check' : 'cross'}
+            sx={{ mr: 1, height: '24px' }}
+          />
+          {isLoading ? 'Loading...' : isActivated ? 'Active' : 'Inactive'}
+        </Typography>
+      </ConditionalWrapper>
+      <Typography variant={'body1Italic'} sx={{ my: 1 }}>
+        Required to satisfy certain access requirements, and recommended for
+        overall account security.
+      </Typography>
+      <Typography variant={'body1'} sx={{ my: 2 }}>
+        Synapse uses a time-based system, which you can set up using an app like
+        Google Authenticator, Duo Mobile, Microsoft Authenticator, or Authy.
+        Certain data may require 2FA to be turned on for your account in order
+        to request access.
+      </Typography>
+      <Stack direction={'row'} gap={1}>
+        <ConditionalWrapper condition={isLoadingStatus} wrapper={Skeleton}>
+          <Button
+            variant={'outlined'}
+            disabled={isLoading}
+            onClick={() => {
+              if (!isActivated) {
+                onBeginTwoFactorEnrollment()
+              } else {
+                disable2FA()
+              }
+            }}
+          >
+            {isActivated ? 'Deactivate' : 'Activate'} 2FA
+          </Button>
+        </ConditionalWrapper>
+        {status?.status !== 'DISABLED' && (
+          <ConditionalWrapper condition={isLoadingStatus} wrapper={Skeleton}>
+            <Button
+              variant={'text'}
+              disabled={isLoading}
+              onClick={() => {
+                onRegenerateBackupCodes()
+              }}
+            >
+              Regenerate Backup Codes
+            </Button>
+          </ConditionalWrapper>
+        )}
+        <Button
```

Comment:
I think we should hide this until the docs link is valid (I don't want this sneaking out into production).